File: app/src/main/java/com/github/ykiselev/playground/init/WindowTask.java
package com.github.ykiselev.playground.init;

import com.github.ykiselev.playground.app.window.AppWindow;

public final class WindowTask extends FrameTask {

    private final AppWindow window;

    public WindowTask(Runnable next, AppWindow window) {
        super(next);
        this.window = window;
    }

    @Override
    protected void onFrameStart() {
        window.makeCurrent();
        window.checkEvents();
    }

    @Override
    protected void onFrameEnd() {
        window.swapBuffers();
    }
}

File: app/src/main/kotlin/com/github/ykiselev/playground/init/WindowTask.kt
package com.github.ykiselev.playground.init

import com.github.ykiselev.playground.app.window.AppWindow

class WindowTask(next: Runnable?, private val window: AppWindow) : FrameTask(next) {

    override fun onFrameStart() {
        window.makeCurrent()
        window.checkEvents()
    }

    override fun onFrameEnd() {
        window.swapBuffers()
    }
}
